### PR TITLE
update Role based calls to new api version to support dataActions

### PR DESCRIPTION
**Pull request by darshanhs90**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by darshanhs90** [comment]
@mcardosos what can be done to remove the semantic issue for this PR,as the path mentioned in the error isn't used anywhere before
Reason for /roleassignmentId and /roledefinitionid to be different is as per the example for each
**/roleassignmentid example:**
id=/subscriptions/<somesubid>/providers/Microsoft.Authorization/roleAssignments/fa1a4d3b-2cca-406b-8956-6b6b32377641
 
**/roledefinitionid example:** id=/subscriptions/<subId>/providers/Microsoft.Authorization/roleDefinitions/f0885aa8-0107-4e65-9a00-541286195838

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-ClassicAdminCalls.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        "operationId": "ClassicAdministrators_List",
+        "description": "Gets service administrator, account administrator, and co-administrators for the subscription.",
+        "parameters": [
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "The API version to use for this operation."
+          },
```
Comment (mcardosos) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-ClassicAdminCalls.json]: Why not use the API version parameter already defined in the parameters section?

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json [hunk 2]
```diff
@@ -0,0 +1,1237 @@
...
+        "parameters": [
+          {
+            "name": "resourceProviderNamespace",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The namespace of the resource provider."
+          },
+          {
+            "name": "api-version",
```
Comment (mcardosos) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json]: >api-version [](start = 21, length = 11)

Same here. If the parameter is already defined somewhere else why not use it?

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json [hunk 3]
```diff
@@ -0,0 +1,1237 @@
...
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Authorization/providerOperations/{resourceProviderNamespace}": {
+      "get": {
+        "tags": [
+          "ProviderOperationsMetadata"
+        ],
+        "operationId": "ProviderOperationsMetadata_Get",
```
Comment (mcardosos) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json]: >ProviderOperationsMetadata_Get [](start = 24, length = 30)

Odata extension can be useful here

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json [hunk 4]
```diff
@@ -0,0 +1,1237 @@
...
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Authorization/providerOperations": {
+      "get": {
+        "tags": [
+          "ProviderOperationsMetadata"
+        ],
+        "operationId": "ProviderOperationsMetadata_List",
```
Comment (mcardosos) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json]: >ProviderOperationsMetadata_List [](start = 24, length = 31)

Odata extension is useful here too

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json [hunk 5]
```diff
@@ -0,0 +1,1237 @@
...
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Authorization/permissions": {
+      "get": {
+        "tags": [
+          "Permissions"
+        ],
+        "operationId": "Permissions_ListForResourceGroup",
+        "description": "Gets all permissions the caller has for a resource group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
```
Comment (mcardosos) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json]: >resourceGroupName [](start = 21, length = 17)

If resource group name parameter is used in many operation, it also can be defined in the parameter section

**Comment by mcardosos** [comment]
Hello @darshanhs90  !
The main issue is that swagger does not allow to define the same path more than once (for swagger `/{roleassignmentid}` and `/{roledefinitionid }` are equivalents.
[x-ms-paths](https://github.com/stankovski/AutoRest/blob/master/Documentation/swagger-extensions.md#x-ms-paths) can be used to avoid that.

**Comment by mcardosos** [comment]
On the other hand, [we are moving away from current branch](https://github.com/Azure/azure-rest-api-specs/wiki/December-2017-Refactoring).

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json [hunk 6]
```diff
@@ -0,0 +1,1237 @@
...
+            "description": "OK - Returns an array of role assignments.",
+            "schema": {
+              "$ref": "#/definitions/RoleAssignmentListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/RoleAssignmentFilter",
```
Comment (mcardosos) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json]: >RoleAssignmentFilter [](start = 37, length = 20)

Not just `RoleAssignment` ?

Comment (darshanhs90) [reply specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json]: no it takes in the filter parameters as query parameter in the url,hence not the roleassignment

---
In reply to: [164927815](https://github.com/Azure/azure-rest-api-specs/pull/2356#discussion_r164927815) [](ancestors = 164927815)

**Comment by darshanhs90** [comment]
@mcardosos when i added the x-ms-path,i got an error regarding that also
{
  "type": "Error",
  "code": "XmsPathsMustOverloadPaths",
  "message": "Paths in x-ms-paths must overload a normal path in the paths section, i.e. a path in the x-ms-paths must either be same as a path in the paths section or a path in the paths sections followed by additional parameters.",

Not sure what it expects in x-ms-path

**Comment by mcardosos** [comment]
@darshanhs90 It expects that paths in the x-ms-path section also have a query (a dumb query ), just for disambiguation/

**Comment by darshanhs90** [comment]
@mcardosos does autorest expect me to have both /roleassignment and /roledefinition in paths and add another /roledefintion?dummy in x-ms paths?

**Comment by mcardosos** [comment]
No, just /roleassignment in paths, and /roledefinition with dummy in x-ms-paths

**Comment by darshanhs90** [comment]
@mcardosos included as mentioned by you,but still the linter is showing an error in this PR

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json (hunk 5 above)
Comment (mcardosos) [reply specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json]: Solved

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json (hunk 2 above)
Comment (mcardosos) [reply specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleBasedCalls.json]: Solved

**Comment by mcardosos** [comment]
Since there are new operations, also adding @ravbhatnagar for review

**Comment by mcardosos** [comment]
On the other hand, we are [moving away from current branch](https://github.com/Azure/azure-rest-api-specs/wiki/December-2017-Refactoring), please poke @marstr if you need help migrating the PR

**Review by mcardosos — APPROVED** [review]

**Comment by mcardosos** [comment]
Sorry @ravbhatnagar , I thought there were new operations, but that is not the case :)

**Pull request merged**